daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Node.scala

```diff
@@ -20,11 +20,15 @@ object Node {
       extends Product
       with Serializable
       with CidContainer[GenNode[Nid, Cid]] {
-    def version: TransactionVersion
+
+    private[lf] def updateVersion(version: TransactionVersion): GenNode[Nid, Cid]
 
     def foreach2(fNid: Nid => Unit, fCid: Cid => Unit): Unit
 
-    private[lf] def updateVersion(version: TransactionVersion): GenNode[Nid, Cid]
+    def optVersion: Option[TransactionVersion] = this match {
```

Comment:
I’m slightly split about this. It makes a few call sites a bit more convenient but I think it might be preferable to be explicit and instead match on the node type in those cases. wdyt @nickchapman-da @remyhaemmerle-da?

Reply:
funnily enough, I just made a commit that moves slightly in the opposite direction.
I reinstated `Node.version` (removing `Node.optVersion`) 
```
    /** minimum version for a transaction containing this node */
    def version: TransactionVersion
```
With implementation for `NodeRollback`
```
    def version: TransactionVersion = TransactionVersion.minExceptions
```

This cleans up and avoid duplicating logic at multiple call sites.

I think this is nicer. But I can remove it if you hate it!

Reply:
I’d prefer to stick to the previous solution or to drop it like I suggested above. The rollback version is really different. E.g., consider some code which wants to validate that the children are never newer than their parent. This does not work with this implementation (well it works now since there is nothing newer than `minExceptions` but it will fail later) so I’d rather have the caller make a deliberate choice on how they want to handle this case than silently do the wrong thing.

Reply:
I reverted this change. I kept the solution with `optVersion`